tests/test.rs

```diff
@@ -398,6 +398,6 @@ fn bytes() {
     let data = b"abc\0123";
     let s = serialize(&data, Infinite).unwrap();
     let s2 = serialize(&Bytes::new(data), Infinite).unwrap();
-    assert_eq!(s, s2);
+    assert_ne!(s, s2);
 }
 
```

Comment:
Looking at this some more, I think that my request is out-of-scope for this PR.  However, the structures serde::bytes::Bytes and serde::bytes::ByteBuf have *special handling* that we certainly want to be testing.  I'll file another issue for this.